src/registry/registry-utils.ts

```diff
@@ -61,7 +57,7 @@ export function getRegistry(
           uri,
           branch,
           logger: childLogger,
-          proxyUrl: enableProxy && isCanonicalRepoUrl(uri) ? PROXY_DEPLOYED_URL : undefined,
+          proxyUrl: enableProxy ? PROXY_DEPLOYED_URL : undefined,
```

Comment:
The intention of isCanonicalRepoUrl() was for extra protection by preventing callers that enable proxy but use a non-hyperlane registry url. This extra "protection" is unnecessary.

Reply:
Does the proxy not work for arbitrary repos? If so why not?

Reply:
design decision to prevent others from using our API key to make arbitrary requests